types/react-table/index.d.ts

```diff
@@ -1,746 +1,673 @@
-export type ExpandedChangeFunction = (column: any, event: any, isTouch: boolean) => void;
-export type ResizedChangeFunction = (newResized: Resize[], event: any) => void;
-export type SortFunction = (a: any, b: any, desc: any) => number;
-
-export interface Resize {
-    id: string;
-    value: any;
+// TypeScript Version: 3.5
+// reflects react-table@7.0.0-rc.9
+
+// tslint:disable:no-unnecessary-generics
+// no-unnecessary-generics is disabled because many of these definitions are either used in a generic
+// context or the signatures are required to match for declaration merging
+
+import { ComponentType, DependencyList, EffectCallback, MouseEvent, ReactElement, ReactNode, RefObject } from 'react';
+
+export {};
+
+/**
+ * The empty definitions of below provides a base definition for the parts used by useTable, that can then be extended in the users code.
+ *
+ * @example
+ *  export interface TableOptions<D extends object = {}}>
+ *    extends
+ *      UseExpandedOptions<D>,
+ *      UseFiltersOptions<D> {}
+ * see https://gist.github.com/ggascoigne/646e14c9d54258e40588a13aabf0102d for more details
+ */
+export interface TableOptions<D extends object> extends UseTableOptions<D> {}
+
+export interface TableInstance<D extends object = {}>
+    extends Omit<TableOptions<D>, 'columns' | 'pageCount'>,
+        UseTableInstanceProps<D> {}
+
+export interface TableState<D extends object = {}> {
+    hiddenColumns?: Array<IdType<D>>;
 }
-export interface Filter {
-    id: string;
-    value: any;
-    pivotId?: string;
+
+export interface Hooks<D extends object = {}> extends UseTableHooks<D> {}
+
+export interface Cell<D extends object = {}> extends UseTableCellProps<D> {}
+
+export interface Column<D extends object = {}> extends UseTableColumnOptions<D> {}
+
+export interface ColumnInstance<D extends object = {}>
+    extends Omit<Column<D>, 'id' | 'columns'>,
+        UseTableColumnProps<D> {}
+
+export interface HeaderGroup<D extends object = {}> extends ColumnInstance<D>, UseTableHeaderGroupProps<D> {}
+
+export interface Row<D extends object = {}> extends UseTableRowProps<D> {}
+
+/* #region useTable */
+export function useTable<D extends object = {}>(
+    options: TableOptions<D>,
+    ...plugins: Array<PluginHook<D>>
+): TableInstance<D>;
+
+/**
+ * NOTE: To use custom options, use "Interface Merging" to add the custom options
+ */
+export type UseTableOptions<D extends object> = {
+    columns: Array<Column<D>>;
+    data: D[];
+} & Partial<{
+    initialState: Partial<TableState<D>>;
+    reducer: (newState: TableState<D>, action: string, previousState: TableState<D>) => TableState<D>;
+    useControlledState: (state: TableState<D>) => TableState<D>;
+    defaultColumn: Partial<Column<D>>;
+    initialRowStateKey: IdType<D>;
+    getSubRows: (originalRow: D, relativeIndex: number) => D[];
+    getRowId: (originalRow: D, relativeIndex: number) => IdType<D>;
```

Comment:
To match type of selectedRowIds: `Set<string>`

```suggestion
    getRowId: (originalRow: D, relativeIndex: number) => string;
```